float32 speed
float32 pitch
float32 heading